src/content/tools/pub/dependencies.md

```diff
@@ -474,6 +474,49 @@ If you modify the pubspec and update to a new major version,
 then you might encounter breaking changes,
 so you need to test even more thoroughly.
 
+### Test with downgraded dependencies
+
+When developing packages for publication, it is often preferable to
+allow as wide dependency constraints as possible.
+A wide dependency constraint reduces the likelihood that
+package consumers face a version resolution conflict.
+
+Hence, if you have a dependency on `foo: ^1.2.3` and
+version `1.3.0` of `foo` is released, it might be reasonable to
+keep the existing dependency constraint (`^1.2.3`).
+But if your package starts using features that were added in `1.3.0`, then
+you'll need to bump your constraint to `^1.3.0`.
+
+However, it's easy to forget to bump a
+dependency constraint when it becomes necessary.
+Therefore, it's a best practice to test your package
+against downgraded dependencies before publishing.
+
+Test against downgraded dependencies by running:
+
+ * [`dart pub downgrade`][],
+ * `dart analyze`, and possibly,
+ * `dart test`.
```

Comment:
Perhaps this can be better formatted? Perhaps with some more context about the steps?

````suggestion
To test against downgraded dependencies, run [`dart pub downgrade`][] and
verify your package still analyzes without errors and passes all tests:

```console
dart pub downgrade
dart analyze
dart test
```
````

\cc @MaryaBelanger For thoughts.

Reply:
I agree!